pmd-core/src/main/java/net/sourceforge/pmd/cli/PMDParameters.java

```diff
@@ -203,6 +215,23 @@ public void validate(String name, Integer value) throws ParameterException {
         }
     }
 
+    public static class PathToRelativizeRootValidator implements IValueValidator<List<Path>> {
+        @Override
+        public void validate(String name, List<Path> value) throws ParameterException {
+            for (Path p : value) {
+                if (Files.isRegularFile(p)) {
+                    throw new ParameterException("Expected a directory path for option " + name + ", found a file: " + p);
+                }
+            }
+        }
+    }
+
+    public static class StringToPathConverter implements IStringConverter<Path> {
+        @Override
+        public Path convert(String value) {
+            return Paths.get(value);
+        }
+    }
```

Comment:
I just realize these should be deprecated in master and removed in PMD 7

Reply:
the whole class is actually `@Deprecated` + `@InternalApi`. not sure, if we need to mark the nested classes as well?

Reply:
not from a public API standpoint, as long as we remember to remove these even if unflagged

Reply:
ok, then we can leave it as is. I've created #4349 for these cases - we have many not yet cleaned up things like that - planned for rc3.